src/Compilers/CSharp/Portable/Symbols/Source/SourceExtensionTypeSymbol.cs

```diff
@@ -40,9 +41,13 @@ public ExtensionInfo(TypeSymbol? underlyingType, ImmutableArray<NamedTypeSymbol>
 
         internal override bool IsExtension => true;
 
+        internal override bool IsExplicitExtension
+            => ((ExtensionDeclarationSyntax)this.declaration.Declarations[0].SyntaxReference.GetSyntax()).ImplicitOrExplicitKeyword
+                .IsKind(SyntaxKind.ExplicitKeyword);
```

Comment:
Going to syntax every time is undesirable, especially in IDE scenarios. We could cache the value in ```declaration.Declarations[0]```, I think there is already a bunch of flags inferred from syntax there.